src/big_text.rs

```diff
@@ -92,18 +98,29 @@ impl Widget for BigText<'_> {
 
 /// Chunk the area into as many x*y cells as possible returned as a 2D iterator of `Rect`s
 /// representing the rows of cells. The size of each cell depends on given font size
-fn layout(
+fn layout<'a>(
     area: Rect,
     pixel_size: &PixelSize,
-) -> impl IntoIterator<Item = impl IntoIterator<Item = Rect>> {
+    alignment: Alignment,
+    lines: &'a [Line<'a>],
+) -> impl IntoIterator<Item = impl IntoIterator<Item = Rect>> + 'a {
     let (step_x, step_y) = pixel_size.pixels_per_cell();
     let width = 8_u16.div_ceil(step_x);
     let height = 8_u16.div_ceil(step_y);
 
     (area.top()..area.bottom())
         .step_by(height as usize)
-        .map(move |y| {
-            (area.left()..area.right())
+        .enumerate()
+        .map(move |(i, y)| {
```

Comment:
Given that the enumerate is only used to index into `lines`, how about using Iter::zip or the Itertools::zip macro instead?

e.g.:
```
        .zip(lines.iter())
        .map(move |(y, line)| {
```